shared/utils/relabel-ids.py

```diff
@@ -73,123 +287,22 @@ def create_xccdf_id_to_cce_id_mapping(xccdftree):
-    # Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1230
-    # Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1229
-    # Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1228
-
-    ovalrules = ovaltree.findall(".//{%s}definition" % oval_ns)
-    for rule in ovalrules:
-        ovalid = rule.get("id")
-        ovaldesc = rule.find(".//{%s}description" % oval_ns)
-        # Assign <reference> CCE elements only to those OVAL checks whose
-        # XCCDF rule ID has <ident> CCE set
-        if ovalid is not None and \
-           ovaldesc is not None and \
-           ovalid in idmappingdict:
-            xccdfcceid = idmappingdict[ovalid]
-            # IF CCE ID IS IN VALID FORM (either 'CCE-XXXX-X' or 'CCE-XXXXX-X'
-            # where each X is a digit, and the final X is a check-digit)
-            if re.search(r'CCE-\d{4,5}-\d', xccdfcceid) is not None:
-                # Then append the <reference source="CCE" ref_id="CCE-ID" /> element right
-                # after <description> element of specific OVAL check
-                ccerefelem = ElementTree.Element(
-                    'reference', ref_id="%s" % xccdfcceid, source="CCE")
-                ovaldesc.addnext(ccerefelem)
-                # Sanity check if appending succeeded
-                if ccerefelem.getprevious() is not ovaldesc:
-                    sys.stderr.write("ERROR: Failed to add CCE ID to %s. "
-                                     "Exiting" % (ovalid))
-                    sys.exit(1)
-
-
-def ensure_by_xccdf_referenced_oval_def_is_defined_in_oval_file(xccdftree, ovaltree, indexed_oval_defs):
-    # Ensure all OVAL checks referenced by XCCDF are implemented in OVAL file
-    # Drop the reference from XCCDF to OVAL definition if:
-    # * Particular OVAL definition isn't present in OVAL file,
-    # * That OVAL definition doesn't constitute a remote OVAL
-    #   (@href of <check-content-ref> doesn't start with 'http'
-    # Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1092
-    # Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1095
-    # Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1098
-
-    for rule in xccdftree.findall(".//{%s}Rule" % xccdf_ns):
-        xccdfid = rule.get("id")
-        if xccdfid is None:
-            # TODO: We probably should issue a warning here or even error..
-            continue
-
-        # Search OVAL ID in OVAL document
-        ovalid = indexed_oval_defs.get(xccdfid)
-        if ovalid is not None:
-            # The OVAL check was found, we can continue
+    for xccdfid, rule in rules_with_ids_generator(xccdftree):
+        identcce = _find_identcce(rule)
+        if identcce is None:
             continue
 
-        # Search same ID in XCCDF document
-        check = None
-        for el in rule.findall(".//{%s}check" % (xccdf_ns)):
-            if el.get("system") != oval_ns:
```

Comment:
I believe that this is a typo and `oval_cs` should be used instead of `oval_ns`. However, this functionality is almost a duplicate of the former `get_checkfiles` function.